src/detectors/BTOF/BTOF.cc

```diff
@@ -64,32 +62,46 @@ void InitPlugin(JApplication *app) {
     } catch(const std::runtime_error&) {
         // Nothing
     }
-    for (auto qualifier : std::vector<std::string>({"", "Seeded"})) {
-        app->Add(new JOmniFactoryGeneratorT<PIDLookup_factory>(
-              fmt::format("CombinedTOF{}LUTPID", qualifier),
-              {
-              fmt::format("Reconstructed{}ChargedWithPFRICHPIDParticles", qualifier),
-              fmt::format("Reconstructed{}ChargedWithPFRICHPIDParticleAssociations", qualifier),
-              },
-              {
-              fmt::format("Reconstructed{}ChargedWithPFRICHTOFPIDParticles", qualifier),
-              fmt::format("Reconstructed{}ChargedWithPFRICHTOFPIDParticleAssociations", qualifier),
-              fmt::format("CombinedTOF{}ParticleIDs", qualifier),
-              },
-              {
-                .filename="calibrations/tof.lut",
-                .system=BarrelTOF_ID,
-                .pdg_values={11, 211, 321, 2212},
-                .charge_values={1},
-                .momentum_edges={0.0, 0.3, 0.6, 0.9, 1.2, 1.5, 1.8, 2.1, 2.4, 2.7, 3.0, 3.3, 3.6, 3.9, 4.2, 4.5, 4.8, 5.1, 5.4, 5.7, 6.0},
-                .polar_edges={2.50, 10.95, 19.40, 27.85, 36.30, 44.75, 53.20, 61.65, 70.10, 78.55, 87.00, 95.45, 103.90, 112.35, 120.80, 129.25, 137.70, 146.15, 154.60},
-                .azimuthal_binning={0., 360., 360.}, // lower, upper, step
-                .momentum_bin_centers_in_lut=true,
-                .polar_bin_centers_in_lut=true,
-              },
-              app
-              ));
-    }
+    PIDLookupConfig pid_cfg {
+      .filename="calibrations/tof.lut",
+      .system=BarrelTOF_ID,
+      .pdg_values={11, 211, 321, 2212},
+      .charge_values={1},
+      .momentum_edges={0.0, 0.3, 0.6, 0.9, 1.2, 1.5, 1.8, 2.1, 2.4, 2.7, 3.0, 3.3, 3.6, 3.9, 4.2, 4.5, 4.8, 5.1, 5.4, 5.7, 6.0},
+      .polar_edges={2.50, 10.95, 19.40, 27.85, 36.30, 44.75, 53.20, 61.65, 70.10, 78.55, 87.00, 95.45, 103.90, 112.35, 120.80, 129.25, 137.70, 146.15, 154.60},
```

Comment:
(Optional) Lots of magic and opportunities for typos in these numbers. Maybe we need a `linspace` helper (https://godbolt.org/z/jd657W4qK):
```cpp
    auto linspace = [](std::floating_point auto start, std::floating_point auto stop, size_t num){
        std::vector<decltype(start)> v(num);
        std::ranges::generate(v, [&start, n = 0, diff = (stop - start)/(num-1)]() mutable {
            return start + n++ * diff;
        });
        return std::move(v);
    };
```

Reply:
TBH, I'd rather not. Most numbers came directly from DSC's, they should do the updates, if they want to.